src/Symfony/Bundle/FrameworkBundle/Secret/FilesSecretStorage.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Secret;
+
+use Symfony\Bundle\FrameworkBundle\Exception\SecretNotFoundException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class FilesSecretStorage implements MutableSecretStorageInterface
+{
+    private const FILE_SUFFIX = '.bin';
+
+    private $secretsFolder;
+    private $encoder;
+    private $filesystem;
+
+    public function __construct(string $secretsFolder, EncoderInterface $encoder)
+    {
+        $this->secretsFolder = rtrim($secretsFolder, '\\/');
+        $this->encoder = $encoder;
+        $this->filesystem = new Filesystem();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function listSecrets(bool $reveal = false): iterable
+    {
+        if (!$this->filesystem->exists($this->secretsFolder)) {
+            return [];
+        }
+
+        foreach ((new Finder())->in($this->secretsFolder)->depth(0)->name('*'.self::FILE_SUFFIX)->files() as $file) {
+            $name = $file->getBasename(self::FILE_SUFFIX);
+            yield $name => $reveal ? $this->getSecret($name) : null;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSecret(string $name): string
+    {
+        $filePath = $this->getFilePath($name);
+
+        if (!is_file($filePath)) {
+            throw new SecretNotFoundException($name);
+        }
+
+        return $this->encoder->decrypt(file_get_contents($filePath));
```

Comment:
file_get_contents can still return false which should throw a SecretNotFoundException. so this is not race-condition free.